src/kostentraeger/edifact/parser.ts

```diff
@@ -0,0 +1,347 @@
+/** based on documents: 
+ *  - Pflege, Technische Anlage 1, Anhang 5: Kostenträgerdatei
+ *  - Sonstige Leistungserbringer, Anlage 1, Anhang 3, Kapitel 10: Kostenträgerdatei
+ * 
+ * (see /docs/documents.md for more info)
+ */
+
+import { parseDate } from "../../edifact/parse_utils"
+import { Interchange, Message } from "../../edifact/types"
+import { 
+    abrechnungscodeSonderschluessel, KostentraegerAbrechnungscodeSchluessel,
+    anschriftartSchluessel, AnschriftartSchluessel,
+    bundeslandSchluessel, BundeslandSchluessel,
+    datenlieferungsArtSchluessel, DatenlieferungsartSchluessel,
+    dfuProtokollSchluessel, DFUProtokollSchluessel,
+    ikVerknuepfungsartSchluessel, IKVerknuepfungsartSchluessel, 
+    kvBezirkSchluessel, KVBezirkSchluessel, 
+    leistungserbringergruppeSchluessel, LeistungserbringergruppeSchluessel, 
+    pflegeLeistungsartSonderschluessel, KostentraegerPflegeLeistungsartSchluessel, 
+    uebermittlungsmediumParameterSchluessel, UebermittlungsmediumParameterSchluessel, 
+    uebermittlungsmediumSchluessel, UebermittlungsmediumSchluessel,
+    uebermittlungszeichensatzSchluessel, UebermittlungszeichensatzSchluessel,
+    uebertragungstageSchluessel, UebertragungstageSchluessel,
+    verarbeitungskennzeichenSchluessel, VerarbeitungskennzeichenSchluessel
+} from "./codes"
+import {
+    abrechnungscodeSchluessel
+} from "../../sgb-v/codes"
+import {
+    leistungsartSchluessel as pflegeLeistungsartSchluessel
+} from "../../sgb-xi/codes"
+import { KTORInterchange, KTORMessage, ANS, ASP, DFU, FKT, IDK, KTO, NAM, UEM, VDT, VKG } from "./segments"
+
+/** 
+ *  Parses a Kostenträgerdatei (payer file) which provides information how to send invoices to the
+ *  statutory health insurance as a health care service provider.
+ * 
+ *  The implementation only covers Pflegedienstleistungen (care providers) and Sonstige Leistungserbringer
+ *  (miscellaneous providers of health care services)
+ */
+
+/* Transforms the serial data (array of arrays) from a Kostenträger EDIFACT-interchange into legible
+ * messages with labelled typesafe segments. It does however not divorce the data structure from 
+ * (the limitations of) the EDIFACT message format yet. */
+export default function parse(interchange: Interchange): KTORInterchange {
+    const header = interchange.header
+    return {
+        spitzenverbandIK: header[1][0],
+        creationDate: parseDate(header[3][0], header[3][1]),
+        institutions: interchange.messages.map((message) => parseMessage(message))
+        // header[6] would contain the file name. Though there is probably no meaning in parsing that
+    }
+}
+
+/** Parse only one message of the interchange */
+function parseMessage(message: Message): KTORMessage {
+    const messageId = parseInt(message.header[0][0])
+    const messageTxt = `Message ${messageId} -`
+    const messageType = message.header[1][0]
+    if (messageType != "KOTR") {
```

Comment:
Did I understand correctly: the message type is called KTOR, but its identifier is KOTR?

Reply:
Ah no, it looks like I very consistently made the same typo. Will fix this.